rosbag2_tests/test/rosbag2_tests/record_fixture.hpp

```diff
@@ -69,6 +69,19 @@ class RecordFixture : public TemporaryDirectoryFixture
     rclcpp::shutdown();
   }
 
+  void wait_for_metadata()
+  {
+    // Waits until metadata is written to disk since some platforms have slower IO sync times
+    // If the metadata does not exist, the test will timeout and fail
+    const auto metadata_path = rcpputils::fs::path{root_bag_path_} / "metadata.yaml";
+    while (true) {
+      if (metadata_path.exists()) {
+        return;
+      }
+      std::this_thread::sleep_for(50ms);
+    }
```

Comment:
This will block forever if the file never gets written.

Reply:
Is it better to have the test timeout with `colcon test` or specifying our own timeout?
This was left as an infinite loop since we expect the `colcon test` to fail on timeout.

Reply:
Nit: specify in the docstring that you expect the timeout to happen as a result of colcon test.

Reply:
You could easily insert a parameter `max_checks` and rewrite `while (true)` with `for (int i = 0; i < max_checks; ++i)` 
Elsewhere have `constexpr const int kDefaultMaxChecks` assigned to 60s / 50ms.